src/dart-2.md

```diff
@@ -5,23 +5,41 @@ title: Dart 2 Migration
 description: How Dart 2 is different from Dart 1.x, and how you can convert your code to work with Dart 2.
 ---
 
-**This page is under construction.**
-It aims to help developers migrating from Dart 1.x to Dart 2.
+Here's the general process for migrating to Dart 2,
+from either Dart 1.x or an earlier version of Dart 2:
 
+1. Get an up-to-date version of the Flutter or Dart SDK.
+   * [Flutter SDK instructions][Flutter SDK install]
+   * [Dart SDK instructions][Dart SDK install] (VM or web)
+2. Upgrade the packages your app depends on.
+   * Flutter: [`flutter packages upgrade`][flutter package upgrade]
+   * Dart VM or web: [`pub upgrade`][pub upgrade]
+3. Run the [dart2_fix tool.][dart2_fix]
+4. Run the analyzer to find [compile-time errors][Fixing Common Type Problems].
+   * Flutter: [`flutter analyze`][Flutter analyzer]
+   * Dart VM or web: [`dartanalyzer`][dartanalyzer] with
+     [Dart 2 semantics][enable strong mode]
+5. Fix issues in your code and run the analyzer again,
+   repeating until your code passes static analysis.
+6. Run tests to find runtime errors.
+   * Run all [automated tests] for your software.
+   * Do manual testing, and look for console errors.
+   Consider adding automated tests to catch issues that you find.
+7. Fix issues until your code works.
 
-## Differences
+Each time the SDK has a significant release, repeat the process.
 
-### Tools
+The rest of this page gives an overview of differences between Dart 1.x and Dart 2,
+with links to where you can find more information.
 
-* Tools related to web development have changed.
-  For details, see the
-  [Dart 2 page for web developers.][webdev dart2]
-  * Pub no longer supports `build` and `serve`.
-  * Dartium is no longer supported.
-* Pub no longer supports transformers.
-  Instead, use the new build system. More information:
-  * [Web-specific instructions for building, serving, and testing using build_runner][build_runner web]
-  * [Build system documentation](https://github.com/dart-lang/build/tree/master/docs)
+{% comment %}
+TODO:
+- talk about library vs. app migration?
+- talk more about versioning?
+{% endcomment %}
+
+
+## Differences
 
 ### Language and libraries
 
```

Comment:
I missed this earlier: from a few lines below ...

    * [Email: Breaking Change: --preview-dart-2 turned on by default][Leaf's email]

The `--` gets converted to an m-dash when the markdown is process. Maybe code-format it?

* [Email: Breaking Change: `--preview-dart-2` turned on by default][Leaf's email]

[Leaf's email]: #

Reply:
Oh! I hadn't noticed this. I'll fix that.